app/models/spree/variant.rb

```diff
@@ -256,6 +257,10 @@ def ensure_unit_value
       self.unit_value = 1.0
     end
 
+    def ensure_shipping_category
+      self.shipping_category ||= DefaultShippingCategory.find_or_create
+    end
```

Comment:
I'm wondering what the need for this is? 

If shipping category was required on products before, then they should all be set. Of course, maybe there's some old records that didn't have one somehow, and that problem would be migrated from the product to variant. But we didn't need to handle the problem before, so why now?